internal/types/compare.go

```diff
@@ -254,3 +258,103 @@ func compareNumbers(a float64, b int64) CompareResult {
 
 	return CompareResult(bigA.Cmp(bigB))
 }
+
+// CompareArrays compares indices of a filter array according to indices of a document array;
+// returns Equal when a document array contains another array(subarray) that equals filter array.
+func CompareArrays(filterArr, docArr *Array) CompareResult {
+	if docArr.Len() == 0 && filterArr.Len() == 0 {
+		return Equal
+	}
+	if filterArr.Len() == 0 {
+		return Incomparable
+	}
+
+	EntireArrayResult, subArrayEquality := Incomparable, Incomparable
+
+	for i := 0; i < docArr.Len(); i++ {
+		arrValue := must.NotFail(docArr.Get(i))
+		switch arrValue := arrValue.(type) {
+		case *Array:
+			filterArrValue := must.NotFail(filterArr.Get(i))
+			switch filterArrValue := filterArrValue.(type) {
+			case *Array:
+				res := CompareArrays(filterArrValue, arrValue)
+				res = handleSubArrayComparingResult(i, &res, &EntireArrayResult, &subArrayEquality)
```

Comment:
Sorry - it's not clear. 
First, we compare filter value and array - what do we get? what for? 
Then it's understandable name,  elements of the array are compared. But the result is overwritten by the following statement, isn't it? Can we do here an early exit: if one of the elements is not Equal, then return?

Reply:
we get a value from `doc array` by an index; if the value is another array(subarray) then we should iterate through it in case the subarray might fit filter array; but what if we got a value from `filter array` by the index it is another array (filter sub array) - then we comparing by this `filterSubArray` so to say. And I don't think earlier return/exit is possible bsc there is a chance on last itterattion we might get sub array that might fit filter array.